x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts

```diff
@@ -0,0 +1,951 @@
+    parent: {
+      id: normalizeValue<string>(fields[PARENT_ID]),
+    },
+    observer: {
+      type: normalizeValue<string>(fields[OBSERVER_TYPE]),
+      version: normalizeValue<string>(fields[OBSERVER_VERSION]),
+      version_major: normalizeValue<number>(fields[OBSERVER_VERSION_MAJOR]),
+    },
+    agent: {
+      name: normalizeValue<AgentName>(fields[AGENT_NAME]),
+      version: normalizeValue<string>(fields[AGENT_VERSION]),
+    },
+    trace: {
+      id: normalizeValue<string>(fields[TRACE_ID]),
+    },
+    '@timestamp': normalizeValue<string>(fields[AT_TIMESTAMP]),
+    service: {
+      name: normalizeValue<string>(fields[SERVICE_NAME]),
+      environment: normalizeValue<string>(fields[SERVICE_ENVIRONMENT]),
+    },
+    event: {
+      outcome: normalizeValue<EventOutcome>(fields[EVENT_OUTCOME]),
+    },
+    processor: {
+      name: normalizeValue<'transaction'>(fields[PROCESOR_NAME]),
+      event: normalizeValue<'span'>(fields[PROCESSOR_EVENT]),
+    },
+    transaction: {
+      id: normalizeValue<string>(fields[TRANSACTION_ID]),
+    },
+    span: {
+      duration: {
+        us: normalizeValue<number>(fields[SPAN_DURATION]),
+      },
+      subtype: normalizeValue<string>(fields[SPAN_SUBTYPE]),
+      name: normalizeValue<string>(fields[SPAN_NAME]),
+      destination: {
+        service: {
+          resource: normalizeValue<string>(fields[SPAN_DESTINATION_SERVICE_RESOURCE]),
+        },
+      },
+      id: normalizeValue<string>(fields[SPAN_ID]),
+      type: normalizeValue<string>(fields[SPAN_TYPE]),
+    },
+    timestamp: {
+      us: normalizeValue<number>(fields[TIMESTAMP]),
+    },
+  };
+};
+
+export const spanLinksDetailsMapping = (fields: Partial<Record<string, unknown[]>>) => {
+  return {
+    trace: {
+      id: normalizeValue<string>(fields[TRACE_ID]),
+    },
+    span: {
+      id: normalizeValue<string>(fields[SPAN_ID]),
+      name: normalizeValue<string>(fields[SPAN_NAME]),
+      type: normalizeValue<string>(fields[SPAN_TYPE]),
+      subtype: normalizeValue<string>(fields[SPAN_SUBTYPE]),
+      duration: {
+        us: normalizeValue<number>(fields[SPAN_DURATION]),
+      },
+    },
+    transaction: {
+      id: normalizeValue<string>(fields[TRANSACTION_ID]),
+      name: normalizeValue<string>(fields[TRANSACTION_NAME]),
+      duration: {
+        us: normalizeValue<string>(fields[TRANSACTION_DURATION]),
+      },
+    },
+    service: {
+      name: normalizeValue<string>(fields[SERVICE_NAME]),
+      environment: normalizeValue<string>(fields[SERVICE_ENVIRONMENT]),
+    },
+    processor: {
+      event: normalizeValue<string>(fields[PROCESSOR_EVENT]),
+    },
+    agent: {
+      name: normalizeValue<AgentName>(fields[AGENT_NAME]),
+    },
+  };
+};
+
+export const linkedParentsOfSpanMapping = (fields: Partial<Record<string, unknown[]>>) => {
+  return {
+    span: {
+      links: [
+        {
+          trace: {
+            // todo(milosz): confirm `span.links.trace.id` format
```

Comment:
span links can potentially be an array of many. `span.links.trace.id` and `span.links.span.id` will need to be merged and iterated over and added to `span.links[]`. Below is a screenshot showing their structure in discover.
<img width="945" alt="Screenshot 2024-09-23 at 08 22 48" src="https://github.com/user-attachments/assets/8af90194-a18c-4c14-bc76-d11866588bf3">

Reply:
How did you get those? I see only single values in `span.links.span.id` and `span.links.trace.id` when running synthtrace scenario (`span_links.ts`). Did you make some aggregations?

However, even for single values in `span.links.span.id` and `span.links.trace.id` indeed the mapping is incorrect because we expect an array of `{
  trace: { id: string };
  span: { id: string };
}` so we can get multiple values. With that in mind, I'm not sure how this data will be represented in `fields` property.

My understanding is that in `source` fields we might get:
```
{
  span: {
    link: [
      { trace: { id: 'aaa', span: { id: 'aaa' } } },
      { trace: { id: 'bbb', span: { id: 'bbb' } } },
      { trace: { id: 'ccc', span: { id: 'ccc' } } },
    ],
  },
}
```

So what format does expect in `fields` in this situation? I guess this is the problem with all array of object properties, I'm not sure how this is solved 😅